cmd/cstor-pool-mgmt/pool/v1alpha2/delete.go

```diff
@@ -0,0 +1,45 @@
+package v1alpha2
+
+import (
+	"github.com/golang/glog"
+	api "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha2"
+	zfs "github.com/openebs/maya/pkg/zfs/cmd/v1alpha1"
+)
+
+// Delete will destroy the pool for given csp.
+// It will also perform labelclear for pool disk.
+func Delete(csp *api.CStorNPool) error {
```

Comment:
Block devices are used during the delete process. But, there are chances that BDs are released as soon as CSPC is deleted.
So, ordering of deletions triggered with 'OwnerReference' is needed, i.e., CSPs, pool deployments, and then BDCs.
This can be achieved by having finalizers on CSPC, and second finalizer on BDC as well.
cc: @AmitKumarDas @kmova @akhilerm @sonasingh46

Reply:
We will rename this to Destroy